compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -610,6 +618,52 @@ public void visit(BLangInvocation invocationExpr) {
         }
     }
 
+    private boolean isFieldsInitializedForSelfKeyWordExpression(BLangInvocation invocationExpr) {
+        if (invocationExpr.expr != null && invocationExpr.expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF
+                && ((BLangSimpleVarRef) invocationExpr.expr).getVariableName().value.equals(Names.SELF.value)) {
```

Comment:
Better if the literal (Names.SELF.value) is on the left side of the equals comparison to avoid possible NPEs.